comps/llms/text-generation/predictionguard/predictionguard_llm.py

```diff
@@ -1,64 +1,57 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+# TODO: Add streaming
 import os
 from fastapi.responses import StreamingResponse
-from langsmith import traceable
-from comps import GeneratedDoc, LLMParamsDoc, ServiceType, opea_microservices, register_microservice
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
 from predictionguard import PredictionGuard
-import os
-import requests
-from langsmith import traceable
+from comps import GeneratedDoc, ServiceType, opea_microservices, register_microservice
+
+# Set up Prediction Guard API key
+PG_API_KEY = os.getenv("PREDICTIONGUARD_API_KEY")
+
+# Define the input data model
+class LLMParamsDoc(BaseModel):
+    model: str = "Neural-Chat-7B"
+    query: str
+    max_new_tokens: int
+    temperature: float
+    top_p: float
+    top_k: int
+    #stream: bool = False
 
-# Initialize Prediction Guard API URL and Key
-PG_API_URL = "https://api.predictionguard.com/completions"
-PG_API_KEY = os.getenv("PREDICTION_GUARD_API_KEY")
+app = FastAPI()
 
 @register_microservice(
     name="opea_service@llm_predictionguard",
     service_type=ServiceType.LLM,
-    endpoint="/v1/chat/completions",
+    endpoint="/v1/completions",
     host="0.0.0.0",
     port=9000,
 )
-@traceable(run_type="llm")
 def llm_generate(input: LLMParamsDoc):
-    headers = {
-        "Authorization": f"Bearer {PG_API_KEY}",
-        "Content-Type": "application/json"
-    }
+    if not PG_API_KEY:
+        raise HTTPException(status_code=500, detail="PREDICTIONGUARD_API_KEY environment variable is not set")
```

Comment:
The prediction guard client already raises an error if the environment variable is not set. This is not needed.